### PR TITLE
Add ability to render R code blocks with RCall

### PR DESCRIPTION
**Pull request by jkrumbiegel**
Code blocks with `{r}` undergo a simple source transformation which adds code that checks that the `RCall` symbol is available and points to the expected module. If that's the case, the whole cell (minus cell options) is treated as a string to be used inside the `RCall.R_str` macro, the return value of which is `RCall.rcopy`d to the Julia side again, to be used as the cell output.

For the code output we have to be a bit tricky to get R syntax highlighting. The actual code cell is always going to be treated as julia because the ipynb format doesn't support multiple languages. But we can insert a simple markdown cell with an R code block into which we copy the source code minus cell options. The code with cell options remains in the code cell but that one gets the `echo: false` option added so it's not visible. But this makes cell annotations relating to the output work, like figure or table captions.

Here's an example script:

````markdown
---
engine: julia
---

This is an R code block but RCall has not been made available in the session, yet, so it will fail.

```{r}
x <- 1
```

Here we make RCall available, `import RCall` would also work

```{julia}
using RCall

x = 123
```

This is R code accessing a variable from Julia

```{r}
a <- 5 * $x
```

Here we have a plot in R with a more complex interpolation

```{r}
#| label: fig-rplot
#| fig-cap: A plot done with an implicit R cell
plot($(sin.(0:0.1:4pi)))
```

And here a data.frame in R, which is auto-converted to a Julia `DataFrame`

```{r}
#| label: tbl-dataframe
#| tbl-cap: A data.frame, autoconverted to Julia
numeric_data <- c(1, 2, 3, 4, 5)
boolean_data <- c(TRUE, FALSE, TRUE, FALSE, TRUE)
categorical_data <- c("A", "B", "C", "A", "B")

# Combine the data into a data frame
df <- data.frame(
  Numeric = numeric_data,
  Boolean = boolean_data,
  Categorical = as.factor(categorical_data)
)
```

This cell contains an R error

```{r}
x + y
```

And this is Julia accessing a variable from R

```{julia}
@rget a
```
````

And here a screenshot of the rendered html:

image

### CODE REVIEW

File: src/server.jl [hunk 1]
```diff
@@ -706,16 +719,45 @@ function evaluate_raw_cells!(
...
                         cells,
                         (;
                             id = expand_cell ? string(nth, "_", mth) : string(nth),
                             cell_type = chunk.type,
                             metadata = (;),
-                            source = process_cell_source(
-                                remote.code,
-                                expand_cell ? remote.cell_options : Dict(),
-                            ),
+                            source = process_cell_source(chunk.source, cell_options),
```
Comment (MichaelHatherly) [inline src/server.jl:760]: A couple lines below here is the code that handles inline code evaluation, e.g.

```qmd
Some inline code: `{julia} 1+1`.
```

Should we also be supporting `{r}` here as well?

Comment (jkrumbiegel) [reply src/server.jl:760]: Good idea

File: src/server.jl [hunk 2]
```diff
@@ -799,6 +845,33 @@ function process_cell_source(source::AbstractString, cell_options::Dict = Dict()
     end
 end
 
+function strip_cell_options(source::AbstractString)
+    lines = split(source, "\n")
```
Comment (MichaelHatherly) [inline src/server.jl:849]: Elsewhere for line splitting we've used `eachline(IOBuffer(source); keep = true)`, can this be adjusted to be consistent with those places?

**Comment by MichaelHatherly** [comment]
Just the one comment, everything else looks good.

**Review by MichaelHatherly — APPROVED** [review]

**Pull request merged**